inc/lift/Request.h

```diff
@@ -370,34 +370,37 @@ class Request {
     ssize_t m_bytes_written { 0 };
 
     /**
-     * The timepoint for when the request started -- we want this is set so we can calculate how long the 
+     * The timepoint for when the request started -- we want this is set so we can calculate how long the
      * request took in the event there is a response wait time timeout.
      * We're using a uint64_t since we get the timepoint from libuv and it uses uint64_t.
      */
     uint64_t  m_start_time{0};
 
     /// The total time it took to receive a response -- will only be set once we receive a response.
     std::optional<std::chrono::milliseconds> m_total_time;
-    
+
     /**
      * Bool indicating whether or not onComplete has been called (true) or not (false) so if a request exceeds
      * its response wait time, its on complete handler can be called only once.
      */
     std::atomic_bool m_on_complete_has_been_called{false};
-    
+
     /**
      * Optional milliseconds indicating the response wait time for the request. If it is set, the request's
      * on complete handler will be called even if we have not received a response and the status code will be
      * set to indicate a response wait time timeout.
      */
     std::optional<std::chrono::milliseconds> m_response_wait_time;
-    
+
     /**
      * Optional iterator to the location in the EventLoop's multiset where the corresponding ResponseWaitTimeWrapper.
      * This will only be set if the response wait time is used.
      */
     std::optional<std::multiset<ResponseWaitTimeWrapper>::iterator> m_response_wait_time_set_iterator;
 
+    /// HTTP status code for request -- will either be set from curl response or use default of 504 (indicating a timeout)
+    http::StatusCode m_http_status_code{http::StatusCode::HTTP_504_GATEWAY_TIMEOUT};
```

Comment:
Defaulting to 504 sounds confusing. Why not a version of `0` so it's differentiated from a real HTTP status code?

Reply:
I like it!